luarules/gadgets/unit_xp_range_bonus.lua

```diff
@@ -38,17 +44,17 @@ function gadget:GameFrame(n)
             SetUnitMaxRange(unitID,newRange)
         end
     end
+
 	updateList = {}
 end
 
 function GG.requestMaverickExpUpdate(unitID)	
-	local unitDef = UnitDefs[GetUnitDefID(unitID)]
-	if(unitDef.customParams.rangexpscale == nil) then
+	if(XPDefs[attackerDefID] == nil) then
```

Comment:
Is `attackerDefID` defined in this context?

Reply:
Fixed this :)